src/masternodes/mn_rpc.cpp

```diff
@@ -128,35 +199,125 @@ std::vector<CTxIn> GetInputs(UniValue const& inputs) {
     return vin;
 }
 
-static std::vector<CTxIn> GetAuthInputs(CWallet* const pwallet, CTxDestination const& auth, UniValue const& explicitInputs) {
-    std::vector<CTxIn> vin{};
-    if (!explicitInputs.empty()) {
-        return GetInputs(explicitInputs.get_array());
-    } else {
-        std::vector<COutput> vecOutputs;
-        CCoinControl cctl;
-        cctl.m_avoid_address_reuse = false;
-        cctl.m_min_depth = 1;
-        cctl.m_max_depth = 999999999;
-        cctl.matchDestination = auth;
-        cctl.m_tokenFilter = {DCT_ID{0}};
-
-        pwallet->BlockUntilSyncedToCurrentChain();
-        auto locked_chain = pwallet->chain().lock();
-        LOCK(pwallet->cs_wallet);
+boost::optional<CScript> AmIFounder(CWallet* const pwallet) {
+    for(auto const & script : Params().GetConsensus().foundationMembers) {
+        if(IsMine(*pwallet, script) == ISMINE_SPENDABLE)
+            return { script };
+    }
+    return {};
+}
+
+static boost::optional<CTxIn> GetAuthInputOnly(CWallet* const pwallet, CTxDestination const& auth) {
+
+    std::vector<COutput> vecOutputs;
+    CCoinControl cctl;
+    cctl.m_avoid_address_reuse = false;
+    cctl.m_min_depth = 1;
```

Comment:
Could we not look for unconfirmed UTXOs as an auth input? Check it is in mempool and we can trust it will be added to the chain, using unconfirmed UTXOs will also reduce issue at hand and is part of the solution here!